src/main/java/com/redhat/devtools/intellij/quarkus/lsp4ij/LSPEditorWrapper.java

```diff
@@ -0,0 +1,246 @@
+        ApplicationManager.getApplication().runReadAction(() -> {
+            final PsiFile file = PsiDocumentManager.getInstance(project)
+                    .getCachedPsiFile(document);
+            if (file == null) {
+                return;
+            }
+            DaemonCodeAnalyzer codeAnalyzer = DaemonCodeAnalyzer.getInstance(project);
+            HighlightInfo info = ((DaemonCodeAnalyzerImpl) codeAnalyzer).findHighlightByOffset(editor.getDocument(), offset, false);
+            if (info != null) {
+
+                if (codeActionResult != null) {
+                    LanguageServer languageServer = codeActionResult.getLanguageServer();
+                    List<Either<Command, CodeAction>> codeActions = codeActionResult.getCodeActions();
+
+                    for (Either<Command, CodeAction> commandOrCodeAction : codeActions) {
+                        LSPCodeActionIntentionAction quickFix = new LSPCodeActionIntentionAction(commandOrCodeAction,
+                                languageServer);
+                        TextRange textRange = new TextRange(start != -1 ? start : offset, end != -1 ? end : offset);
+                        HighlightDisplayKey k = HighlightDisplayKey.find("Annotator");
+                        info.registerFix(quickFix, null, HighlightDisplayKey.getDisplayNameByKey(k), textRange, k);
+                    }
+
+                    // See UpdateHighlightersUtil
+                    List<Pair<HighlightInfo.IntentionActionDescriptor, RangeMarker>> list =
+                            new ArrayList<>(info.quickFixActionRanges.size());
+                    for (Pair<HighlightInfo.IntentionActionDescriptor, TextRange> pair : info.quickFixActionRanges) {
+                        TextRange textRange = pair.second;
+                        RangeMarker marker = document.createRangeMarker(textRange);
+                        list.add(Pair.create(pair.first, marker));
+                    }
+                    info.quickFixActionMarkers = ContainerUtil.createLockFreeCopyOnWriteList(list);
+                }
+            }
+        });
+    }
+
+    public static boolean hasWrapper(Editor editor) {
+        return editor.getUserData(KEY) != null;
+    }
+
+    public void dispose() {
+        this.editor.removeEditorMouseMotionListener(editorMouseMotionListener);
+        this.editor.getCaretModel().removeCaretListener(caretListener);
+        this.editor.getDocument().removeDocumentListener(documentListener);
+        editor.putUserData(KEY, null);
+    }
+
+    public static LSPEditorWrapper getLSPEditorWrapper(Editor editor) {
+        LSPEditorWrapper wrapper = editor.getUserData(KEY);
+        if (wrapper != null) {
+            return wrapper;
+        }
+        return getLSPEditorWrapperSync(editor);
+    }
+
+    private static synchronized LSPEditorWrapper getLSPEditorWrapperSync(Editor editor) {

```

Comment:
why is this method called `getLSPEditorWrapperSync`? All I see in differences when compared to `getLSPEditorWrapper` is that it would create the wrapper if it doesn't exist. Shouldn't it be called `getOrCreateLSPEditorWrapper`?